sdk/ai/ai-projects/README.md

```diff
@@ -0,0 +1,711 @@
+    - [Create thread](#create-thread) with
+      - [Tool resource](#create-thread-with-tool-resource)
+    - [Create message](#create-message) with:
+      - [File search attachment](#create-message-with-file-search-attachment)
+      - [Code interpreter attachment](#create-message-with-code-interpreter-attachment)
+    - [Execute Run, Create Thread and Run, or Stream](#create-run-run_and_process-or-stream)
+    - [Retrieve message](#retrieve-message)
+    - [Retrieve file](#retrieve-file)
+    - [Tear down by deleting resource](#teardown)
+    - [Tracing](#tracing)
+  - [Tracing](#tracing)
+    - [Installation](#installation)
+    - [Tracing example](#tracing-example)
+- [Troubleshooting](#troubleshooting)
+  - [Exceptions](#exceptions)
+  - [Reporting issues](#reporting-issues)
+  <!-- - [Next steps](#next-steps) -->
+- [Contributing](#contributing)
+
+## Getting started
+
+### Prerequisite
+
+- [LTS versions of Node.js](https://github.com/nodejs/release#release-schedule)
+- An [Azure subscription][azure_sub].
+- A [project in Azure AI Foundry](https://learn.microsoft.com/azure/ai-studio/how-to/create-projects?tabs=ai-studio).
+- The project connection string. It can be found in your Azure AI Foundry project overview page, under "Project details". Below we will assume the environment variable `PROJECT_CONNECTION_STRING` was defined to hold this value.
+- Entra ID is needed to authenticate the client. Your application needs an object that implements the [TokenCredential](https://learn.microsoft.com/javascript/api/@azure/core-auth/tokencredential) interface. Code samples here use [DefaultAzureCredential](https://learn.microsoft.com/javascript/api/@azure/identity/defaultazurecredential?view=azure-node-latest). To get that working, you will need:
+  - The `Contributor` role. Role assigned can be done via the "Access Control (IAM)" tab of your Azure AI Project resource in the Azure portal.
+  - [Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli) installed.
+  - You are logged into your Azure account by running `az login`.
+  - Note that if you have multiple Azure subscriptions, the subscription that contains your Azure AI Project resource must be your default subscription. Run `az account list --output table` to list all your subscription and see which one is the default. Run `az account set --subscription "Your Subscription ID or Name"` to change your default subscription.
+
+### Install the package
+
+```bash
+npm install @azure/ai-projects
+```
+
+## Key concepts
+
+### Create and authenticate the client
+
+The class factory method `fromConnectionString` is used to construct the client. To construct a client:
+
+```javascript
+import { AIProjectsClient } from "@azure/ai-projects";
+import { DefaultAzureCredential } from "@azure/identity";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const connectionString = process.env["AZURE_AI_PROJECTS_CONNECTION_STRING"] || "<connectionString>";
```

Comment:
Prerequisite mentions Entra ID is needed, should we use an example with entra id instead?

Reply:
These are not secrets, it of <Host;subscription;resourcegroup;project name>